src/main/java/wooteco/subway/service/member/FavoriteService.java

```diff
@@ -0,0 +1,60 @@
+package wooteco.subway.service.member;
+
+import org.springframework.stereotype.Service;
+import wooteco.subway.domain.member.Favorite;
+import wooteco.subway.domain.member.Favorites;
+import wooteco.subway.domain.member.Member;
+import wooteco.subway.domain.member.MemberRepository;
+import wooteco.subway.service.exception.SameStationException;
+import wooteco.subway.service.member.dto.FavoriteRequest;
+import wooteco.subway.service.member.dto.FavoriteResponse;
+import wooteco.subway.service.station.StationService;
+
+import java.util.List;
+
+@Service
+public class FavoriteService {
+    private MemberRepository memberRepository;
+    private StationService stationService;
+
+    public FavoriteService(MemberRepository memberRepository, StationService stationService) {
+        this.memberRepository = memberRepository;
+        this.stationService = stationService;
+    }
+
+    public void createFavorite(FavoriteRequest favoriteRequest, Member member) {
+        validate(favoriteRequest);
+        Long sourceId = stationService.findStationByName(favoriteRequest.getSourceName()).getId();
+        Long destinationId = stationService.findStationByName(favoriteRequest.getDestinationName()).getId();
+        member.addFavorite(new Favorite(sourceId, destinationId));
+        memberRepository.save(member);
+    }
+
+    public void removeFavorite(FavoriteRequest favoriteRequest, Member member) {
+        validate(favoriteRequest);
+        Long sourceId = stationService.findStationByName(favoriteRequest.getSourceName()).getId();
+        Long destinationId = stationService.findStationByName(favoriteRequest.getDestinationName()).getId();
+        member.removeFavoriteById(sourceId, destinationId);
+        memberRepository.save(member);
+    }
+
+    public List<FavoriteResponse> showAllFavorites(Member member) {
+        Favorites favorites = new Favorites(member.getFavorites());
+
+        return favorites.toFavoriteResponses(stationService.findStations());
+    }
+
+    public boolean ifFavoriteExist(FavoriteRequest favoriteRequest, Member member) {
```

Comment:
```suggestion
    public boolean hasFavorite(FavoriteRequest favoriteRequest, Member member) {
        validate(favoriteRequest);
        Long sourceId = stationService.findStationByName(favoriteRequest.getSourceName()).getId();
        Long destinationId = stationService.findStationByName(favoriteRequest.getDestinationName()).getId();
        return member.hasFavorite(sourceId, destinationId);
    }
```
member 객체가 favorites 객체를 가지고 있으니 member 통해서 favorite이 있는지 알수 있겠네요
member 객체는 favorites의 메시지 결과를 단순 전달하는 역할만 하면 될거에요